src/test/java/rx/ObservableTests.java

```diff
@@ -1138,4 +1142,53 @@ public void testSubscribingSubscriberAsObserverMaintainsSubscriptionChain() {
 
         subscriber.assertUnsubscribed();
     }
+    
+    @Test
+    public void testExtend() {
+        final TestSubscriber<Object> subscriber = new TestSubscriber<Object>();
+        final Object value = new Object();
+        Observable.just(value).x(new ConversionFunc<Object,Object>(){
+            @Override
+            public Object convert(OnSubscribe<Object> onSubscribe) {
+                onSubscribe.call(subscriber);
+                subscriber.assertNoErrors();
+                subscriber.assertCompleted();
+                subscriber.assertValue(value);
+                return subscriber.getOnNextEvents().get(0);
+            }});
+    }
+
+    @Test
+    public void testOperatorConversion() {
+        final TestSubscriber<Object> subscriber = new TestSubscriber<Object>();
+        final Object value = new Object();
+        Observable.just(value).x(new OperatorConversion<Object,Object>(new Operator<Object, Object>(){
```

Comment:
Per my comments in #2971 I don't think it's advisable to expose the whole Observable type inside the extension. If the operators of Observable are available then you should expect to see flatmaps and groupBys being executed inside a conversion. I do not think that you can achieve the same thing since with `Observable.to` exposes all operators within a conversion step that is meant to subscribe and collect values.